logserver/src/main/java/com/yahoo/logserver/handlers/archive/FilesArchived.java

```diff
@@ -30,21 +28,28 @@
      */
     private final File root;
 
+    private final Object mutex = new Object();
+
     // known-existing files inside the archive directory
     private List<LogFile> knownFiles;
 
-    public final static long compressAfterMillis = 2L * 3600 * 1000;
-    private long maxAgeDays = 30; // GDPR rules: max 30 days
-    private long sizeLimit = 30L * (1L << 30); // 30 GB
+    public static final long compressAfterMillis = 2L * 3600 * 1000;
+    private static final long maxAgeDays = 30; // GDPR rules: max 30 days
+    private static final long sizeLimit = 30L * (1L << 30); // 30 GB
+
+    private void waitForTrigger(long milliS) throws InterruptedException {
+        synchronized (mutex) {
+            mutex.wait(milliS);
+        }
+    }
 
     private void run() {
         try {
-            Thread.sleep(125000); // 2 m 5 s
             while (true) {
                 while (maintenance()) {
```

Comment:
now you do the same anyway, this could be simpler:
```
        while (true) {
                maintenance();
                waitForTrigger(2000);
            }
```

Reply:
Good point.